### PR TITLE
Add explicit anchors links to every anchor

### PR DESCRIPTION
**Pull request by tyhoff**
I ran `yarn run docusaurus write-heading-ids . docs/*` on the whole project and generated these links, and then cleaned up a few ugly ones.

### CODE REVIEW

File: docs/mcu/da1469x-sdk-guide.mdx [hunk 1]
```diff
@@ -255,24 +255,24 @@ void your_ble_task(void) {
 }
 ```
 
-## Verification & Tuning Steps
+## Verification & Tuning Steps {#verification--tuning-steps}
```
Comment (tyhoff) [inline docs/mcu/da1469x-sdk-guide.mdx:258]:
```suggestion
## Verification & Tuning Steps {#verification-tuning-steps}
```

File: docs/mcu/espressif-esp32-guide.mdx [hunk 2]
```diff
@@ -52,9 +52,9 @@ project:
 - [8. Configure Memfault Project Key](#8-configure-memfault-project-key)
 - [9. Regenerate sdkconfig and recompile](#9-regenerate-sdkconfig-and-recompile)
 
-## Detailed Walkthrough
+## Detailed Walkthrough {#detailed-walkthrough}
 
-### 1. Clone Memfault SDK
+### 1. Clone Memfault SDK {#1-clone-memfault-sdk}
```
Comment (tyhoff) [inline docs/mcu/espressif-esp32-guide.mdx:57]:
```suggestion
### 1. Clone Memfault SDK {#clone-memfault-sdk}
```

File: docs/mcu/espressif-esp32-guide.mdx [hunk 3]
```diff
@@ -74,7 +74,7 @@ so it is tracked as part of the project's git repository:
 git submodule add https://github.com/memfault/memfault-firmware-sdk.git third-party/memfault-firmware-sdk
 ```
 
-### 2. Add Memfault SDK to the ESP-IDF CMake Build System
+### 2. Add Memfault SDK to the ESP-IDF CMake Build System {#2-add-memfault-sdk-to-the-esp-idf-cmake-build-system}
```
Comment (tyhoff) [inline docs/mcu/espressif-esp32-guide.mdx:77]:
```suggestion
### 2. Add Memfault SDK to the ESP-IDF CMake Build System {#add-memfault-sdk-to-the-esp-idf-cmake-build-system}
```

File: docs/mcu/espressif-esp32-guide.mdx [hunk 4]
```diff
@@ -110,7 +110,7 @@ add_custom_command(TARGET ${CMAKE_PROJECT_NAME}.elf
   python ${MEMFAULT_FIRMWARE_SDK}/scripts/fw_build_id.py ${CMAKE_PROJECT_NAME}.elf)
 ```
 
-### 3. Add the Memfault Configuration Files
+### 3. Add the Memfault Configuration Files {#3-add-the-memfault-configuration-files}
```
Comment (tyhoff) [inline docs/mcu/espressif-esp32-guide.mdx:113]:
```suggestion
### 3. Add the Memfault Configuration Files {#add-the-memfault-configuration-files}
```

File: docs/mcu/espressif-esp32-guide.mdx [hunk 5]
```diff
@@ -165,7 +165,7 @@ Quotes on filenames and paths without spaces are not required in CMake lists.
 
 :::
 
-### 4. Enable Coredump Collection in sdkconfig
+### 4. Enable Coredump Collection in sdkconfig {#4-enable-coredump-collection-in-sdkconfig}
```
Comment (tyhoff) [inline docs/mcu/espressif-esp32-guide.mdx:168]:
```suggestion
### 4. Enable Coredump Collection in sdkconfig {#enable-coredump-collection-in-sdkconfig}
```

File: docs/mcu/espressif-esp32-guide.mdx [hunk 6]
```diff
@@ -192,7 +192,7 @@ Note: The remainder of the `COREDUMP` settings, including
 `CONFIG_ESP_COREDUMP_DATA_FORMAT_ELF`, do not affect the Memfault coredump. They
 are specific to the ESP-IDF coredump features.
 
-### 5. Create a Coredump Flash Partition
+### 5. Create a Coredump Flash Partition {#5-create-a-coredump-flash-partition}
```
Comment (tyhoff) [inline docs/mcu/espressif-esp32-guide.mdx:195]:
```suggestion
### 5. Create a Coredump Flash Partition {#create-a-coredump-flash-partition}
```

File: docs/mcu/espressif-esp32-guide.mdx [hunk 7]
```diff
@@ -239,7 +239,7 @@ partition CSV file". Optionally, you can edit the sdkconfig file directly
 
 <ProjectAwareCreateAProject />
 
-### 6. Implement `memfault_platform_get_device_info()`
+### 6. Implement `memfault_platform_get_device_info()` {#6-implement-memfault_platform_get_device_info}
```
Comment (tyhoff) [inline docs/mcu/espressif-esp32-guide.mdx:242]:
```suggestion
### 6. Implement `memfault_platform_get_device_info()` {#implement-memfault_platform_get_device_info}
```

File: docs/mcu/espressif-esp32-guide.mdx [hunk 8]
```diff
@@ -282,7 +282,7 @@ Note: there is a non-zero chance of overlapping addresses from Espressif, so it
 is not recommended for large production fleets of devices. As many as 20
 duplicates in a set of 4000 chips have been reported in ESP-IDF user forums.
 
-### 7. Initializing Memfault
+### 7. Initializing Memfault {#7-initializing-memfault}
```
Comment (tyhoff) [inline docs/mcu/espressif-esp32-guide.mdx:285]:
```suggestion
### 7. Initializing Memfault {#initializing-memfault}
```

File: docs/mcu/espressif-esp32-guide.mdx [hunk 9]
```diff
@@ -306,7 +306,7 @@ void app_main() {
 When using SDK versions prior to 0.40.0, an additional step is required. You
 will need to explicitly set `CONFIG_MEMFAULT_AUTOMATIC_INIT=n` in your project.
 
-### 8. Configure Memfault Project Key
+### 8. Configure Memfault Project Key {#8-configure-memfault-project-key}
```
Comment (tyhoff) [inline docs/mcu/espressif-esp32-guide.mdx:309]:
```suggestion
### 8. Configure Memfault Project Key {#configure-memfault-project-key}
```

File: docs/mcu/espressif-esp32-guide.mdx [hunk 10]
```diff
@@ -328,7 +328,7 @@ In ESP-IDF this can be set in a few different places:
   idf.py build -DSDKCONFIG_DEFAULTS="sdkconfig.defaults;sdkconfig.memfault_project_key"
   ```
 
-### 9. Regenerate sdkconfig and recompile
+### 9. Regenerate sdkconfig and recompile {#9-regenerate-sdkconfig-and-recompile}
```
Comment (tyhoff) [inline docs/mcu/espressif-esp32-guide.mdx:331]:
```suggestion
### 9. Regenerate sdkconfig and recompile {#regenerate-sdkconfig-and-recompile}
```

File: docs/mcu/event-serialization-overview.mdx [hunk 11]
```diff
@@ -84,7 +84,7 @@ handwritten because no standard C implementation for embedded devices exists.
 Even if we minify (strip the whitespace from) the example above, the event
 message payload winds up being 471 bytes!
 
-### MessagePack / CBOR
+### MessagePack / CBOR {#messagepack--cbor}
```
Comment (tyhoff) [inline docs/mcu/event-serialization-overview.mdx:87]:
```suggestion
### MessagePack / CBOR {#messagepack-cbor}
```

File: docs/mcu/heap-stats.mdx [hunk 12]
```diff
@@ -53,15 +53,15 @@ wrappers
   </TabItem>
   <TabItem value="baremetal">
 
-### 1. Enable the config flag
+### 1. Enable the config flag {#1-enable-the-config-flag}
```
Comment (tyhoff) [inline docs/mcu/heap-stats.mdx:56]:
```suggestion
### 1. Enable the config flag {#enable-the-config-flag}
```

File: docs/mcu/heap-stats.mdx [hunk 13]
```diff
@@ -53,15 +53,15 @@ wrappers
...
 
 Set this in `memfault_platform_config.h`:
 
 ```c
 #define MEMFAULT_COREDUMP_COLLECT_HEAP_STATS 1
 ```
 
-### 2. Instrument the `malloc` and `free` calls
+### 2. Instrument the `malloc` and `free` calls {#2-instrument-the-malloc-and-free-calls}
```
Comment (tyhoff) [inline docs/mcu/heap-stats.mdx:64]:
```suggestion
### 2. Instrument the `malloc` and `free` calls {#instrument-the-malloc-and-free-calls}
```

File: docs/platform/jira-integration.mdx [hunk 14]
```diff
@@ -25,12 +25,12 @@ You should see the screen below.
...
 In each field, input the following:
 
-### Instance URL
+### Instance URL {#instance-url}
 
 This is the URL of the Jira instance. If it is hosted by Jira, it is likely
 _https://company-name.atlassian.net/_
 
-### Parameters for `/secure/CreateIssueDetails!init.jspa`
+### Parameters for `/secure/CreateIssueDetails!init.jspa` {#parameters-for-securecreateissuedetailsinitjspa}
```
Comment (tyhoff) [inline docs/platform/jira-integration.mdx:33]:
```suggestion
### Parameters for `/secure/CreateIssueDetails!init.jspa` {#parameters}
```

File: docs/mcu/zephyr-guide.mdx [hunk 15]
```diff
@@ -195,7 +195,7 @@ The Memfault SDK is configured in two places:
    If these files are not provided by the user, they will be silently ignored.
    See the Kconfig options for some control over how these files are included.
 
-## Zephyr `__ASSERT()`
+## Zephyr `__ASSERT()` {#zephyr-__assert}
```
Comment (tyhoff) [inline docs/mcu/zephyr-guide.mdx:198]:
```suggestion
## Zephyr `__ASSERT()` {#zephyr-assert}
```

**Review by noahp — APPROVED** [review]
:fire: (can you add the script in a comment or something? would be sweet to run in ci)

**Pull request merged**